### DISCUSSION TITLE
Firebase distribution fails when building from yaml

### DISCUSSION THREAD

**Discussion by eukevin** [codemagic-ci-cd/codemagic-docs#1115] [Q&A] [answered]
After switching our workflows from using the workflow editor to using the codemagic.yaml files, our builds fail at the Firebase publishing step with the following error:

```Error: Failed to authenticate, have you run firebase login?```

The working workflow configuration was exported and only minor changes have since been made. The relevant credentials appear to be  present

```
      firebase:
        firebase_service_account: Encrypted(...)
```

Has anyone run into this issue before?

**Comment by icarusdust** [comment]
Hi, let's pass your service account json file as an env var as follows:

```
  firebase:
   firebase_service_account: $FIREBASE_SERVICE_ACCOUNT
```
More info [here](https://docs.codemagic.io/yaml-publishing/firebase-app-distribution/)

**Reply by eukevin** [reply]
Thank you, that did the trick!

**Reply by eukevin** [reply]
My apologies, in my excitement I accidentally ran the build using the workflow editor. Re-running it using the YAML with your suggestion included yielded the same error as before.

**Comment by icarusdust** [comment]
Did you create an env var and transfer the group that holds your env var to yaml ? As a note, your json file does not need to be base64 encoded.

**Comment by eukevin** [comment]
Yes. The variable was added with variable group ```firebase``` and the following added to the yaml:
```
environment:
  groups:
    - firebase
 ```

**Reply by icarusdust** [reply]
At this point, there must be sth wrong with the provided json value. I strongly suggest you to create a new service account json file and try with it. If not result, send me a build id please.

**Reply by eukevin** [reply]
Will do, thank you.

**Reply by eukevin** [reply]
This has solved the issue, thanks for your help!

**Comment by icarusdust** [comment, marked as answer]
At this point, there must be sth wrong with the provided json value. I strongly suggest you to create a new service account json file and try with it. If not result, send me a build id please.
